lightning/src/routing/network_graph.rs

```diff
@@ -1050,13 +1101,19 @@ impl NetworkGraph {
 				if msg.flags & 1 == 1 {
 					dest_node_id = channel.node_one.clone();
 					if let Some((sig, ctx)) = sig_info {
-						secp_verify_sig!(ctx, &msg_hash, &sig, &channel.node_two);
+						secp_verify_sig!(ctx, &msg_hash, &sig, &PublicKey::from_slice(&channel.node_two.0).map_err(|_| LightningError{
+							err: "Couldn't parse source node pubkey".to_owned(),
+							action: ErrorAction::IgnoreAndLog(Level::Debug)
+						})?);
 					}
 					maybe_update_channel_info!(channel.two_to_one, channel.node_two);
 				} else {
 					dest_node_id = channel.node_two.clone();
 					if let Some((sig, ctx)) = sig_info {
-						secp_verify_sig!(ctx, &msg_hash, &sig, &channel.node_one);
+						secp_verify_sig!(ctx, &msg_hash, &sig, &PublicKey::from_slice(&channel.node_one.0).map_err(|_| LightningError{
+							err: "Couldn't parse destination node pubkey".to_owned(),
+							action: ErrorAction::IgnoreAndLog(Level::Debug)
+						})?);
```

Comment:
Could we assume `PublicKey::from_slice` always returns `Ok` if we constructed the `NodeId` from a `PublicKey`?